src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidator.java

```diff
@@ -399,7 +399,7 @@ private void checkOneSwitchFlowConflict(FlowEndpoint source, FlowEndpoint destin
 
     @VisibleForTesting
     void checkDiverseFlow(RequestedFlow targetFlow) throws InvalidFlowException {
-        if (targetFlow.getSrcSwitch().equals(targetFlow.getDestSwitch())) {
+        if (targetFlow.isOneSwitchFlow()) {
             throw new InvalidFlowException("Couldn't add one-switch flow into diverse group",
```

Comment:
Actually you can add one switch flow into a diverse group. Yes, long flow wouldn't be rerouted immediately but, it long flow will be rerouted later - it will try to stay away from one switch flow. So I think this `if` can be removed

Reply:
I was thinking, that such behavior would confuse a user, but delayed rerouting works fine, so, fixed. Thank you!